internal/iruntime/total_memory_linux.go

```diff
@@ -26,16 +26,32 @@ const unlimitedMemorySize = 9223372036854771712
 // TotalMemory returns total available memory.
 // This implementation is meant for linux and uses cgroups to determine available memory.
 func TotalMemory() (uint64, error) {
-	cgroups, err := cgroups.NewCGroupsForCurrentProcess()
+	var memoryQuota int64
+	var defined bool
+	var err error
+
+	isV2, err := cgroups.IsCGroupV2()
 	if err != nil {
 		return 0, err
 	}
-	memoryQuota, defined, err := cgroups.MemoryQuota()
-	if err != nil || !defined {
-		return 0, err
+
+	if isV2 {
+		memoryQuota, defined, err = cgroups.MemoryQuotaV2()
+		if err != nil {
+			return 0, err
+		}
+	} else {
+		cgroups, err := cgroups.NewCGroupsForCurrentProcess()
```

Comment:
nit: name the var something other than cgroups since that name collides with an import.